google/resource_dataproc_cluster.go

```diff
@@ -0,0 +1,936 @@
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+
+					if len(value) > 55 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 55 characters", k))
+					}
+					if !regexp.MustCompile("^[a-z0-9-]+$").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q can only contain lowercase letters, numbers and hyphens", k))
+					}
+					if !regexp.MustCompile("^[a-z]").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q must start with a letter", k))
+					}
+					if !regexp.MustCompile("[a-z0-9]$").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q must end with a number or a letter", k))
+					}
+					return
+				},
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "global",
+				ForceNew: true,
+			},
+
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+				// GCP automatically adds a 'goog-dataproc-cluster-name' label
+				Computed: true,
+			},
+
+			"cluster_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"delete_autogen_bucket": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"staging_bucket": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						// If the user does not specify a staging bucket, GCP will allocate one automatically.
+						// The staging_bucket field provides a way for the user to supply their own
+						// staging bucket. The bucket field is purely a computed field which details
+						// the definitive bucket allocated and in use (either the user supplied one via
+						// staging_bucket, or the GCP generated one)
+						"bucket": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"gce_cluster_config": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+
+									"zone": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Computed: true,
+										ForceNew: true,
+									},
+
+									"network": {
+										Type:          schema.TypeString,
+										Optional:      true,
+										Computed:      true,
+										ForceNew:      true,
+										ConflictsWith: []string{"cluster_config.gce_cluster_config.subnetwork"},
+										StateFunc: func(s interface{}) string {
+											return extractLastResourceFromUri(s.(string))
```

Comment:
I think this will be ok, but it does make me a bit nervous to only store the end portion of the network since hypothetically it could be in a different project. Do other resources have good ways to handle the name vs url problem? I took a quick look and didn't find anything great, but maybe you'll have better luck :) (I think we usually just use a diffsuppress)